### ISSUE TITLE
Questions: training embeddings from scratch and usage of offline triplets

### ISSUE THREAD

**Issue by fralik** [KevinMusgrave/pytorch-metric-learning#30]
Hi!

I would like to learn embeddings for my set of triplets without a pre-trained network. I am wondering if this is possible with the library and what is the right direction.
I looked at #14 and I have the following questions:

## Architecture

If I write my own architecture, shall I then stop at embedding layer? Something like this:
```python
import torch
import torch.nn as nn


class Flatten(nn.Module):
    def forward(self, x):
        N, C, H, W = x.size()  # read in N, C, H, W
        return x.view(N, -1)


class ConvNet(nn.Module):
    def __init__(self):
        super(ConvNet, self).__init__()
        # input size (1, 256, 256)
        self.model = nn.Sequential(
            # size change from (1, 256, 256) to (64, 256, 256)
            nn.Conv2d(in_channels=1, out_channels=64, kernel_size=5, padding=2),
            nn.LeakyReLU(inplace=True),
            # size change from (64, 256, 256) to (64, 128, 128)
            nn.MaxPool2d(kernel_size=2),
            Flatten(),
            # here, 2048 are my embeddings
            nn.Linear(in_features=64*128*128, out_features=2048),
        )
        
    def forward(self, x):
        output = self.model(x)

        return output
        
```

## Labels
You mention that loss functions need labels. I have a set of predefined triplets, but they do not have any hard labels, i.e. I can not say that two anchors belong to the same class.

As I understood, I must use a sampler. Perhaps, `samplers.FixedSetOfTriplets`, but it  needs labels. Shall I somehow make artificial labels?

Thanks!

**Comment by KevinMusgrave** [comment]
@fralik I think your convnet will work.

Re: predefined triplets, will each batch of size N contain N/3 triplets?

**Comment by fralik** [comment]
Thanks for the quick reply! I managed to proceed with N/3 approach.
Just need to figure out that it learns good :)
